web/app/[workspaceSlug]/(projects)/projects/(detail)/[projectId]/views/(list)/header.tsx

```diff
@@ -21,6 +26,26 @@ export const ProjectViewsHeader = observer(() => {
     membership: { currentProjectRole },
   } = useUser();
   const { currentProjectDetails, loader } = useProject();
+  const { filters, updateFilters, clearAllFilters } = useProjectView();
+
+  const handleRemoveFilter = useCallback(
+    (key: keyof TViewFilterProps, value: string | null) => {
+      let newValues = filters.filters?.[key];
+
+      if (key === "favorites") {
+        newValues = !!value;
+      }
+      if (Array.isArray(newValues)) {
+        if (!value) newValues = [];
+        else newValues = newValues.filter((val) => val !== value) as string[];
+      }
+
+      updateFilters("filters", { [key]: newValues });
+    },
+    [filters.filters, updateFilters]
+  );
+
+  const isFiltersApplied = calculateTotalFilters(filters?.filters ?? {}) !== 0;
 
   const canUserCreateIssue =
```

Comment:
Shouldn't this be canUserCreateView, or is it like if the user can create an issue he can create a view as-well?

Reply:
yeah, i'll need to change the naming.